src/main/java/com/demo/classroom/Service/UserService.java

```diff
@@ -0,0 +1,34 @@
+package com.demo.classroom.Service;
+
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import com.demo.classroom.Entity.User;
+import com.demo.classroom.Repository.TeacherRepository;
+import com.demo.classroom.Repository.StudentRepository;
+
+import com.demo.classroom.Repository.UserRepository;
+
+@Service
+public class UserService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+    private final TeacherRepository teacherRepository;
+    private final StudentRepository studentRepository;
+
+    public UserService(UserRepository userRepository, TeacherRepository teacherRepository, StudentRepository studentRepository) {
+        this.userRepository = userRepository;
+        this.teacherRepository = teacherRepository;
+        this.studentRepository = studentRepository;
+    }
```

Comment:
Can we use Autowired?

Reply:
Yes we should use autowired. Its added now/